graphql/resolvers/search.ts

```diff
@@ -12,8 +14,47 @@ interface SearchResultItemConnection {
   };
   nodes: SearchResultItem[];
   filterOptions: AggResults;
+  isCurrentTerm: boolean;
 }
+function determineIfCurrentTerm(
+  maxEndDate: number,
+  resultSearch: SearchResult[],
+  termId: string
+): boolean {
+  // (e.g today is 9/20/22 and the semester ends 8/30/22) => not a current term
+  const date = new Date().getTime();
+  const currentDate = Math.floor(date / 8.64e7);
+  if (maxEndDate != null && maxEndDate != -1) {
+    return maxEndDate > currentDate;
+  }
 
+  for (const result of resultSearch) {
+    if (result.type === "class") {
+      if (
+        result != undefined &&
+        result.sections[0].meetings != undefined &&
+        result.sections[0].meetings[0] != undefined &&
+        result.sections[0].meetings[0].endDate != undefined &&
+        result.sections[0].meetings[0].endDate > maxEndDate
+      ) {
+        maxEndDate = result.sections[0].meetings[0].endDate;
+      }
+      //     result.sections != undefined &&
+      //     result.sections[0].meetings != undefined &&
+      //     result.sections[0].meetings[0].endDate != undefined &&
+      //     result.sections[0].meetings[0].endDate != null &&
+      //     result.sections[0].meetings[0].endDate > maxEndDate
+      //   ) {
+      //
+      //   }
```

Comment:
Read this page about the [optional chaining operator.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) Basically, it allows us to avoid exactly this scenario in a really compact and nice manner

```suggestion
      const endDate = result?.sections[0]?.meetings[0]?.endDate;
      if (endDate !== undefined && endDate > maxEndDate) {
        maxEndDate = endDate;
      }
```